dht/token-sharding.hh

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "dht/token.hh"
+
+namespace dht {
+
+token bias(uint64_t value);      // translate from a zero-baed range
+
+uint64_t unbias(const token& t); // translate to a zero-baed range
+
+unsigned zero_based_shard_of(uint64_t zero_based_token, unsigned shards, unsigned sharding_ignore_msb_bits);
+
+std::vector<uint64_t> init_zero_based_shard_start(unsigned shards, unsigned sharding_ignore_msb_bits);
```

Comment:
`bias`, `unbias`, `zero_based_shard_of`, and `init_zero_based_shard_start` were previously private murmur3_partitioner methods, and they are still an implementation detail after this change.

I suggest dropping them from the header file and making them `static` inside dht/token.cc.

Reply:
Good idea. I moved all of them but init_zero_based_shard_start which is actually called from dht/i_partitioner.cc